src/visual/decal.ts

```diff
@@ -0,0 +1,47 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+// @ts-nocheck
+
+import ExtensionAPI from '../ExtensionAPI';
+import GlobalModel from '../model/Global';
+import {createOrUpdatePatternFromDecal} from '../util/decal';
+
+export default function (ecModel: GlobalModel, api: ExtensionAPI) {
+    ecModel.eachRawSeries(seriesModel => {
+        const data = seriesModel.getData();
+
+        if (seriesModel.useColorPaletteOnData) {
```

Comment:
It should be
```ts
if (data.hasItemVisual()) { ... }
```
Since in the `itemStyle` of each data decal can be set respectively.

`hasItemVisual` is not implemented yet. It should be in the `List` and determine if `_itemVisuals.length > 0`

```ts
hasItemVisual() {
  return this._itemVisuals.length > 0
}
```